File: app/src/main/java/com/dkaluta/prosary/models/StationsCatalog.kt
package com.dkaluta.prosary.models

/** The fixed catalog of all fourteen Stations of the Cross, in order. This is static structural
 * data (which station is 3rd, etc.), not business logic — display text lives in the content
 * layer, keyed by each station's [Station.imageKey]. The Stations equivalent of MysteryCatalog. */
object StationsCatalog {
    val all: List<Station> = listOf(
        Station(order = 1, imageKey = "station_01_condemned_to_death"),
        Station(order = 2, imageKey = "station_02_carries_his_cross"),
        Station(order = 3, imageKey = "station_03_falls_first_time"),
        Station(order = 4, imageKey = "station_04_meets_his_mother"),
        Station(order = 5, imageKey = "station_05_simon_of_cyrene"),
        Station(order = 6, imageKey = "station_06_veronica"),
        Station(order = 7, imageKey = "station_07_falls_second_time"),
        Station(order = 8, imageKey = "station_08_women_of_jerusalem"),
        Station(order = 9, imageKey = "station_09_falls_third_time"),
        Station(order = 10, imageKey = "station_10_stripped_of_garments"),
        Station(order = 11, imageKey = "station_11_nailed_to_the_cross"),
        Station(order = 12, imageKey = "station_12_dies_on_the_cross"),
        Station(order = 13, imageKey = "station_13_taken_down_from_the_cross"),
        Station(order = 14, imageKey = "station_14_laid_in_the_tomb"),
    )
}

File: Windows/Prosary/Models/StationsCatalog.cs
namespace Prosary.Models;

/// <summary>
/// The fixed catalog of all fourteen Stations of the Cross, in order. This is static structural
/// data (which station is 3rd, etc.), not business logic — display text lives in the content
/// layer, keyed by each station's <see cref="Station.ImageKey"/>. The Stations equivalent of
/// <see cref="MysteryCatalog"/>.
/// </summary>
public static class StationsCatalog
{
    public static readonly IReadOnlyList<Station> All =
    [
        new Station(1, "station_01_condemned_to_death"),
        new Station(2, "station_02_carries_his_cross"),
        new Station(3, "station_03_falls_first_time"),
        new Station(4, "station_04_meets_his_mother"),
        new Station(5, "station_05_simon_of_cyrene"),
        new Station(6, "station_06_veronica"),
        new Station(7, "station_07_falls_second_time"),
        new Station(8, "station_08_women_of_jerusalem"),
        new Station(9, "station_09_falls_third_time"),
        new Station(10, "station_10_stripped_of_garments"),
        new Station(11, "station_11_nailed_to_the_cross"),
        new Station(12, "station_12_dies_on_the_cross"),
        new Station(13, "station_13_taken_down_from_the_cross"),
        new Station(14, "station_14_laid_in_the_tomb"),
    ];
}
